include/net/http_parser.h

```diff
@@ -227,6 +230,7 @@ struct http_parser_settings {
 	http_cb      on_chunk_header;
 	http_cb      on_chunk_complete;
 };
+#endif
```

Comment:
Could add the config option in comment here as there are almost 200 lines inside this define block.
So something like this:
#endif /* !CONFIG_HTTP_PARSER_URL_ONLY */